### ISSUE TITLE
PyJWT: ImportError: cannot import name 'ExpiredSignature'

### ISSUE THREAD

**Issue by AndreyMZ** [DefectDojo/django-DefectDojo#3878]
### Bug description
Regression of bug #3556.

### Environment information
DefectDojo Commit Message:
~~~
[2021-02-17 10:19:48 +0100] 45f64b93: Update rabbitmq Docker tag from 3.8.11 to v3.8.12 (docker-compose.yml) (#3874) [ (HEAD, upstream/dev)]
~~~

### Steps to reproduce
Try to log in to DefectDojo.

#### Actual result
~~~
[17/Feb/2021 18:41:05] "GET /login HTTP/1.1" 200 14013
[...]
Internal Server Error: /login
Traceback (most recent call last):
  [...]
  File "C:\GIT\DefectDojo\venv\lib\site-packages\social_core\backends\azuread_tenant.py", line 6, in <module>
    from jwt import DecodeError, ExpiredSignature, decode as jwt_decode
ImportError: cannot import name 'ExpiredSignature'
~~~

### Root cause
[requirements.txt](https://github.com/DefectDojo/django-DefectDojo/blob/45f64b9300a03fe81172540255fe2bbe4096f19a/requirements.txt)
~~~
social-auth-app-django==4.0.0
[...]
PyJWT==2.0.1
~~~

### Additional information
This bug is added by PR #3644.

**Comment by valentijnscholten** [comment]
I was afraid this could happen, the situation is quite unclear. Some libs support only <2 other only >=2 for PyJWT. I think we need someone actually using the Azure login to find out what PR we need to fix it. Maybe just downgrade back to 1.7.1?

@AndreyMZ Are you using Azure login, or are you seeing it in another social login as well? I think we have people running dev with Google SSO for example, without issues.

**Comment by valentijnscholten** [comment]
Should be fixed if we update to `social-auth-core` 4.0.3: https://github.com/python-social-auth/social-core/commit/e26c71cb9ee2fce9a423336ffb9dbc5a62ff4582

**Comment by AndreyMZ** [comment]
> @AndreyMZ Are you using Azure login, or are you seeing it in another social login as well? I think we have people running dev with Google SSO for example, without issues.

Sorry for the delay in reply. Yes, usually I use Azure AD login. But this issue is reproduced without any social auth, because `'social_core.backends.azuread_tenant.AzureADTenantOAuth2'` is in `AUTHENTICATION_BACKENDS` unconditionally (see [dojo/settings/settings.dist.py:358](https://github.com/DefectDojo/django-DefectDojo/blob/052aa78a8bbe820bad09f3cfc61a5c37d1afae70/dojo/settings/settings.dist.py#L363)).

**Comment by valentijnscholten** [comment]
> Sorry for the delay in reply. Yes, usually I use Azure AD login. But this issue is reproduced without any social auth, because `'social_core.backends.azuread_tenant.AzureADTenantOAuth2'` is in `AUTHENTICATION_BACKENDS` unconditionally (see [dojo/settings/settings.dist.py:358](https://github.com/DefectDojo/django-DefectDojo/blob/052aa78a8bbe820bad09f3cfc61a5c37d1afae70/dojo/settings/settings.dist.py#L363)).

That's weird because it works locally here and also in our integration tests. I did see that depending on the python / pip version it sometimes installs 1.7.x and sometimes 2.0.x. Looking at your log info looks like you're also running outside docker, which might explain it. Anyway, with that PR at least the most important libs are now used against 2.0.x.

We may better not include all those auth backend by default, on the other hand, by having them there by default we found out about this issue quite fast.

**Comment by damiencarol** [comment]
> We may better not include all those auth backend by default, on the other hand, by having them there by default we found out about this issue quite fast.

This is it.

**Comment by damiencarol** [comment]
closing as it is fixed in `1.13.0`
